### DISCUSSION TITLE
[10 Giugno 2022] - Modifica CPU

### DISCUSSION THREAD

**Discussion by rimaout** [CS-Swap/Architettura-degli-Elaboratori#27] [Modifica Architettura]
<img width="1115" alt="image" src="https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/116072651/9f673c10-3031-4205-8abf-5421599f1f10">
<img width="901" alt="image" src="https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/116072651/3d01ac63-71d1-4a93-93ba-561b0cb0ce6d"> 

 <details><summary>Soluzione Prof</summary><p> 
<img width="962" alt="image" src="https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/116072651/df6a2ed4-fa56-4661-8822-3b832437c8f4">
<img width="674" alt="image" src="https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/116072651/48d1aa8a-b882-4382-a342-8f67d76be11d">

</p></details>

**Comment by thegeek-sys** [comment]
<img width="1030" alt="Screenshot 2024-06-04 alle 09 38 31" src="https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/55345633/5c4d189a-e3bb-478a-be0e-10a4c1790ba5">
<img width="1073" alt="Screenshot 2024-06-04 alle 09 38 53" src="https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/55345633/2a21a070-0ed9-4ea7-b42d-3ddd384471ec">

**Comment by AglaiaNorza** [comment]
<img width="997" alt="824510BB-0B59-416C-B777-B7F2E4275CBF" src="https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/75022526/5ca8af2e-48d5-4562-9a13-4205fd0afdc2">

versione alternativa senza l'altro mux - visto che l'indirizzo a cui brancha è lo stesso di un beq, aggiungo una or con le condizioni del bltsafe al mux del beq - non serve settare il segnale branch a 0 perché è in AND con il segnale "zero" della ALU, che nel caso della bltsafe non sarà 1, in quanto rt>rs

**Comment by BroLore** [comment]
![image](https://github.com/CS-Swap/Architettura-degli-Elaboratori/assets/123998905/77595c2e-5120-40c1-bdeb-35f5d5315647)
